security/src/main/java/io/airlift/security/utils/CloseableGSSContext.java

```diff
@@ -0,0 +1,52 @@
+package io.airlift.security.utils;
+
+import io.airlift.security.exception.AuthenticationException;
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSException;
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.LoginException;
+
+import java.io.Closeable;
+import java.net.URI;
+import java.net.UnknownHostException;
+
+import static java.util.Objects.requireNonNull;
+
+public class CloseableGSSContext
+        implements Closeable
+{
+    private GSSContext context;
+
+    public CloseableGSSContext(String serviceName, URI serviceURI)
+    {
+        requireNonNull(serviceName, "serviceName is null");
+        requireNonNull(serviceURI, "serviceURI is null");
+        try {
+            String servicePrincipal = KerberosUtil.getServicePrincipal(serviceName, serviceURI.getHost());
+            Subject clientSubject = KerberosUtil.getSubject(null, true);
+            context = KerberosUtil.getGssContext(clientSubject, servicePrincipal, true);
```

Comment:
But the idea is that anyone that needs a GSSContext would use `CloseableGSSContext` in a try-with-resources block, so they wouldn't be accessing that method directly, anyway.